### ISSUE TITLE
How to connect to TestNet?

### ISSUE THREAD

**Issue by Maddest** [JKorf/Binance.Net#754]
Hi,
I've build a wrapper DLL of the Binance API. The following code works fine on the live net, but if I change to test net. I get a NULL exeception while executing the GetExchangeInfo method. I've changed the base address and generated the test net API keys. 

Anybody knows what I'm doing wrong?

`private BinanceClient HydraRestClient { get; set; }

        
        
        //public void InitializeRest(string binanceAPIKey, string binanceAPISecret)
        public void InitializeRest(string binanceAPIKey, string binanceAPISecret)
        {
            BinanceClient.SetDefaultOptions(new BinanceClientOptions
            {
                ApiCredentials = new ApiCredentials(binanceAPIKey, binanceAPISecret)
                , BaseAddress = "https://testnet.binance.vision"

            });

            this.HydraRestClient = new BinanceClient();           
        }

         

        //Public functions

        public HydraExchangeInfo GetExchangeInfo()
        {
            var ExchangeInfo = this.HydraRestClient.Spot.System.GetExchangeInfo();

            List<HydraSymbols> SymbolList = new List<HydraSymbols>();
            foreach (BinanceSymbol bs in ExchangeInfo.Data.Symbols)
            {
                SymbolList.Add(new HydraSymbols() 
                { 
                    Symbol = bs.Name
                    ,BaseCurrency = bs.BaseAsset
                    ,QuoteCurrency = bs.QuoteAsset
                    ,Status = bs.Status.ToString()
                });
            }

            List<HydraRateLimit> RateLimits = new List<HydraRateLimit>();
            foreach(BinanceRateLimit brl in ExchangeInfo.Data.RateLimits)
            {
                RateLimits.Add(new HydraRateLimit
                {
                    Type = (int)brl.Type
                    , Interval = (int)brl.Interval
                    , IntervalNumber = brl.IntervalNumber
                    , Limit = brl.Limit
                });
            }

            HydraExchangeInfo hei = new HydraExchangeInfo()
            {
                TimeZone = ExchangeInfo.Data.TimeZone
                , ServerTime = ExchangeInfo.Data.ServerTime
                , Symbols = SymbolList.ToArray()
                , RateLimits = RateLimits.ToArray()
            };
            
            return hei;
        }`

**Comment by HypsyNZ** [comment]
Can you post the exception and stack trace if possible

`     var ExchangeInfo = this.HydraRestClient.Spot.System.GetExchangeInfo();`

Put a break point on this and check if the request is actually successful or do something like 

```cs
var exchangeInfoResult = client.Spot.System.GetExchangeInfo();
       if (exchangeInfoResult.Success)
         {
                 // Info
         }
             else
         {
                // Error
         }
```

**Comment by Maddest** [comment]
Hi @HypsyNZ ,

thanks a lot for your help. With the help of your reply I have debugged the following error message from GetExchangeInfo:

**The request was aborted: Could not create SSL/TLS secure channel**

This happens only on Binance Testnet, not live net. 

I googled around and found this solution by @jkorf himself: https://github.com/JKorf/Binance.Net/issues/62

I have added ''ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;', to my init function and now it works fine :-).

Thank you very much!!
